### ISSUE TITLE
Feature request: Add CSS Variables or className funtionality

### ISSUE THREAD

**Issue by princemuel** [rishi-raj-jain/astro-font#4]
When using tools like tailwind, it would be great to have a CSS variable or class-name
that we can reference somewhere else, maybe in config or CSS files...similar to the next/font package

**Comment by rishi-raj-jain** [comment]
Hey @princemuel,

Is the `selector` attribute is what you're referring to? 

https://github.com/rishi-raj-jain/astro-font/blob/master/packages/astro-font/utils.ts#L16

With `selector`, you can configure as to on what className does the CSS applies to.

For e.g.,
```
<AstroFont
  config={[
    {
      // font-name to be used when generating the CSS
      name: 'Afacad',
      src: [
        {
          weight: '400',
          style: 'normal',
          path: './public/fonts/Afacad-Regular.ttf',
          // Some other properties to be used with the this particular font
          css: {
            'font-feature-settings': 'normal',
          },
        },
      ],
      // Whether to preload the font in the output HTML
      preload: true,
      // Whether to swap the display
      display: 'swap',
      
      // Where to apply this whole font (using CSS selector)
      selector: '.random_class',
      
      // What kind of fallback to use: serif or sans-serif
      fallback: 'sans-serif',
    },
  ]}
/>
```

**Comment by rishi-raj-jain** [comment]
Updated the README.md to have it:

https://github.com/rishi-raj-jain/astro-font/tree/master?tab=readme-ov-file#configuring-css-classes

**Comment by princemuel** [comment]
Okay. Will try it out. Thanks. Still would like the css variable option

**Comment by rishi-raj-jain** [comment]
@princemuel 

What's the expectation from the css variable option? Like the DX you want to have? Any ideas would be great.

**Comment by rishi-raj-jain** [comment]
@princemuel 

What's the expectation from the css variable option? Like the DX you want to have? Any ideas would be great.

**Comment by rishi-raj-jain** [comment]
What comes to my mind is a CSS like:

:root {
    --some_css_name: 'Inter', font_fallback_number, 'serif'
}

And then var(--some_css_name) can be used as the font-family.

Make sense?

**Comment by princemuel** [comment]
> What comes to my mind is a CSS like:
> 
> :root {
>     --some_css_name: 'Inter', font_fallback_number, 'serif'
> }
> 
> And then var(--some_css_name) can be used as the font-family.
> 
> Make sense?

Exactly 💯

**Comment by rishi-raj-jain** [comment]
Can you test in the latest version released of https://www.npmjs.com/package/astro-font?

**Comment by rishi-raj-jain** [comment]
https://github.com/rishi-raj-jain/astro-font/blob/master/src/pages/index.astro#L49-L51

is how you can do it.

**Comment by rishi-raj-jain** [comment]
Do let me know in case this doesn't work as expected and I'll re-open this.

**Comment by rishi-raj-jain** [comment]
https://github.com/rishi-raj-jain/astro-font?tab=readme-ov-file#configuring-css-variables added the step to the README.

**Comment by princemuel** [comment]
Hiya! So I gave it a spin today, and overall, it's working like a charm. However, I did make a couple of observations that I thought I'd share:

- The CSS variable being set directly to the :root selector is causing it to overwrite the default tailwind font config. It might be beneficial to take a look at this interaction.
- It would be super helpful if the fallback config could support a broader range of options. Currently, it's somewhat limited to serif and sans-serif. Including options like monospace, system-ui etc...(or leaving it totally up to the user??) could greatly enhance the flexibility. I, for instance, was working with a monospace font for code blocks on my site and needed to provide a monospace fallback.

I must say, though, your project is truly impressive and has a lot of potential. I'm open to contributing if there's an opportunity to collaborate and improve upon these aspects. Keep up the great work!
